### ISSUE TITLE
fsr3 frame generation?

### ISSUE THREAD

**Issue by YakovAU** [ndepoel/FSR2Unity#13]
is frame generation supported for dx11 games?

**Comment by ndepoel** [comment]
No, in fact Frame Generation is not a part of this project at all yet. I made it a point to rename the project to "FSR 3.0 Upscaler" to emphasize that it only includes the upscaling part of FSR3.

Frame Generation for Unity is an ongoing research topic and if it ever happens, it will likely be released as an entirely separate project. The biggest issue is that for FG to be fully implemented, a custom swapchain needs to be integrated which is a very low-level part of the rendering pipeline, something that Unity does not expose to game developers even in the scriptable render pipelines. There may be some workarounds for this problem but at the moment it is still uncertain whether it can be done in a way that's viable for commercial product releases, as opposed to just some outside hack that most likely violates Unity's terms of use.
